flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/table/MongoDBTableSource.java

```diff
@@ -0,0 +1,195 @@
+
+/**
+ * A {@link DynamicTableSource} that describes how to create a MongoDB change stream events source
+ * from a logical description.
+ */
+public class MongoDBTableSource implements ScanTableSource {
+
+    private final TableSchema physicalSchema;
+    private final String uri;
+    private final String database;
+    private final String collection;
+    private final Boolean errorsLogEnable;
+    private final String errorsTolerance;
+    private final Boolean copyExisting;
+    private final String copyExistingPipeline;
+    private final Integer copyExistingMaxThreads;
+    private final Integer copyExistingQueueSize;
+    private final Integer pollMaxBatchSize;
+    private final Integer pollAwaitTimeMillis;
+    private final Integer heartbeatIntervalMillis;
+
+    public MongoDBTableSource(
+            TableSchema physicalSchema,
+            String uri,
+            String database,
+            String collection,
+            @Nullable String errorsTolerance,
+            @Nullable Boolean errorsLogEnable,
+            @Nullable Boolean copyExisting,
+            @Nullable String copyExistingPipeline,
+            @Nullable Integer copyExistingMaxThreads,
+            @Nullable Integer copyExistingQueueSize,
+            @Nullable Integer pollMaxBatchSize,
+            @Nullable Integer pollAwaitTimeMillis,
+            @Nullable Integer heartbeatIntervalMillis) {
+        this.physicalSchema = physicalSchema;
+        this.uri = checkNotNull(uri);
+        this.database = checkNotNull(database);
+        this.collection = checkNotNull(collection);
+        this.errorsTolerance = errorsTolerance;
+        this.errorsLogEnable = errorsLogEnable;
+        this.copyExisting = copyExisting;
+        this.copyExistingPipeline = copyExistingPipeline;
+        this.copyExistingMaxThreads = copyExistingMaxThreads;
+        this.copyExistingQueueSize = copyExistingQueueSize;
+        this.pollMaxBatchSize = pollMaxBatchSize;
+        this.pollAwaitTimeMillis = pollAwaitTimeMillis;
+        this.heartbeatIntervalMillis = heartbeatIntervalMillis;
+    }
+
+    @Override
+    public ChangelogMode getChangelogMode() {
+        return ChangelogMode.newBuilder()
+                .addContainedKind(RowKind.INSERT)
+                .addContainedKind(RowKind.UPDATE_AFTER)
+                .addContainedKind(RowKind.DELETE)
+                .build();
+    }
+
+    @Override
+    public ScanRuntimeProvider getScanRuntimeProvider(ScanContext scanContext) {
+        RowType rowType = (RowType) physicalSchema.toRowDataType().getLogicalType();
+        TypeInformation<RowData> typeInfo =
+                scanContext.createTypeInformation(physicalSchema.toRowDataType());
+
+        DebeziumDeserializationSchema<RowData> deserializer =
+                new MongoDBConnectorDeserializationSchema(rowType, typeInfo);
+
+        MongoDBSource.Builder<RowData> builder =
+                MongoDBSource.<RowData>builder()
+                        .connectionUri(uri)
```

Comment:
@Jiabao-Sun There is also the url here. If Shard, there will be multiple ip:ports.

Reply:
Hi. @XuQianJin-Stars 
The uri represents MongoDB's connecting string, you can see https://docs.mongodb.com/manual/reference/connection-string/#standard-connection-string-format for detail.

For replica set.
`mongodb://mongodb1.example.com:27317,mongodb2.example.com:27017/?replicaSet=mySet`

For sharded cluster.
`mongodb://mongos0.example.com:27017,mongos1.example.com:27017,mongos2.example.com:27017`

Reply:
This uri pattern looks a little complicate. I'm afraid most MongoDB newbies can't set a correct URI at first time. 
Can we decouple the different parts of the URI? e.g. `hosts`, `replica-set`, `username`, `password`. 

Decouple can also makes things flexible, e.g. we would like to mask secrets in logging and displaying.

Reply:
Hi @Jiabao-Sun I also think it is better to set the username and password separately.

Reply:
Thanks, I will add added separate configurations for that.

Reply:
@wuchong @XuQianJin-Stars

New config options have been added to replace the `uri`.

- `hosts`:  The comma-separated list of hostname and port pairs of the MongoDB servers. 
- `user`:  Database user to be used when connecting to MongoDB.
- `password`: Password to be used when connecting to MongoDB.
- `mongodb.replicaset`: Specifies the name of the replica set. It is not necessary, but can speed up your connection times to explicitly state the servers configured by `hosts`. 
- `mongodb.authsource`: This is required only when MongoDB is configured to use authentication with another authentication database than admin. Defaults to `admin`.
- `mongodb.connect.timeout.ms`: The time in milliseconds to attempt a connection before timing out. Defaults to 10000 (10 seconds).
- `mongodb.socket.timeout.ms`: The time in milliseconds to attempt a send or receive on a socket before the attempt times out. Defaults to 0.
- `mongodb.ssl.enabled`: Connector will use SSL to connect to MongoDB instances.
- `mongodb.ssl.invalid.hostname.allowed`: When SSL is enabled this setting controls whether strict hostname checking is disabled during connection phase.